src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -86,4 +102,41 @@ public void setServerList(List<BitbucketServerConfiguration> serverList) {
     public boolean hasAnyInvalidConfiguration() {
         return serverList.stream().anyMatch(server -> server.validate().kind == Kind.ERROR);
     }
+
+    private void updateJobs(Map<String, String> oldBaseUrls) {
+        Set<String> changedServerIds = serverList.stream()
+                .filter(serverConfig -> !serverConfig.getBaseUrl().equalsIgnoreCase(oldBaseUrls.get(serverConfig.getId())))
+                .map(BitbucketServerConfiguration::getId)
+                .collect(toSet());
+        if (!changedServerIds.isEmpty()) {
+            try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+                Jenkins.get().getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)
+                        .forEach(job -> {
+                            if (job instanceof AbstractProject && ((AbstractProject) job).getScm() instanceof BitbucketSCM) {
+                                AbstractProject<?, ?> project = (AbstractProject<?, ?>) job;
+                                BitbucketSCM bitbucketSCM = (BitbucketSCM) project.getScm();
+                                if (changedServerIds.contains(bitbucketSCM.getServerId())) {
+                                    try {
+                                        // This server has had its base URL updated so we need to recalculate the clone URL
+                                        project.setScm(new BitbucketSCM(bitbucketSCM));
```

Comment:
Calling `new BitbucketSCM` will force it to go out to Bitbucket to generate the clone URL for the repo and update the UserRemoteConfig